core-java-modules/core-java-lang-oop-patterns/src/test/java/com/baeldung/cloning/DeepCloningTest.java

```diff
@@ -0,0 +1,39 @@
+package com.baeldung.cloning;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.math.BigDecimal;
+
+import org.junit.Test;
+
+public class DeepCloningTest {
+
+    @Test
+    public void whenCloningObject_thenObjectsShouldNotBeTheSame() throws CloneNotSupportedException {
+        Country country = new Country("KE", "Kenya", new Currency("KES", "Ksh", BigDecimal.valueOf(1000)));
+        Country deepClone = country.clone();
+
+        assertThat(country).isNotSameAs(deepClone);
+        assertThat(country.getCurrency()).isNotSameAs(deepClone.getCurrency());
+
```

Comment:
minor, but lets remove this empty line here

Reply:
resolved